frontend/lib/src/components/widgets/DateInput/DateInput.test.tsx

```diff
@@ -107,124 +109,103 @@ describe("DateInput widget", () => {
 
-    expect(wrapper.find(UIDatePicker).prop("value")).toStrictEqual([newDate])
+    expect(screen.getByTestId("stDateInputInput")).toHaveValue(newDate)
     expect(props.widgetMgr.setStringArrayValue).toHaveBeenCalledWith(
       props.element,
-      ["2020/02/06"],
+      [newDate],
       {
         fromUi: true,
       }
     )
   })
 
-  it("reset it's value to default when it closed with empty input", () => {
+  it("resets its value to default when it's closed with empty input", () => {
     const props = getProps()
     jest.spyOn(props.widgetMgr, "setStringArrayValue")
 
-    const wrapper = mount(<DateInput {...props} />)
-    const newDate = new Date("2020/02/06")
+    render(<DateInput {...props} />)
+    const dateInput = screen.getByTestId("stDateInputInput")
 
-    // @ts-expect-error
-    wrapper.find(UIDatePicker).prop("onChange")({
-      date: newDate,
+    fireEvent.change(dateInput, {
+      target: { value: newDate },
     })
-    wrapper.update()
 
-    expect(wrapper.find(UIDatePicker).prop("value")).toStrictEqual([newDate])
-    // @ts-expect-error
-    wrapper.find(UIDatePicker).prop("onChange")({
-      date: null,
+    expect(dateInput).toHaveValue(newDate)
+
+    // Simulating clearing the date input
+    fireEvent.change(dateInput, {
+      target: { value: null },
     })
-    // @ts-expect-error
-    wrapper.find(UIDatePicker).prop("onClose")()
-    wrapper.update()
-    expect(wrapper.find(UIDatePicker).prop("value")).toStrictEqual([
-      new Date("1970/1/20"),
-    ])
+
+    // Simulating the close action
+    fireEvent.blur(dateInput)
+    expect(dateInput).toHaveValue(fullOriginalDate)
   })
 
   it("has a minDate", () => {
+    // The fireEvent.change will modify the input's value unconditionally. This is because
+    // the underlying input element doesn't possess a 'min' attribute.
+    let dateInputRef: any
     const props = getProps()
-    const wrapper = mount(<DateInput {...props} />)
-    expect(wrapper.find(UIDatePicker).prop("minDate")).toStrictEqual(
-      new Date("1970/1/20")
+    render(
+      <DateInput
+        // @ts-expect-error
+        ref={ref => {
+          dateInputRef = ref
+        }}
+        {...props}
+      />
     )
-    expect(wrapper.find(UIDatePicker).prop("maxDate")).toBeUndefined()
+    expect(dateInputRef.props.element.min).toStrictEqual(originalDate)
+    expect(dateInputRef.props.element.max).toBe("")
   })
 
   it("has a maxDate if it is passed", () => {
+    // The fireEvent.change will modify the input's value unconditionally. This is because
+    // the underlying input element doesn't possess a 'max' attribute.
+    let dateInputRef: any
     const props = getProps({ max: "2030/02/06" })
-    const wrapper = mount(<DateInput {...props} />)
-
-    expect(wrapper.find(UIDatePicker).prop("maxDate")).toStrictEqual(
-      new Date("2030/02/06")
-    )
-  })
-
-  it("handles min dates with years less than 100", () => {
```

Comment:
I don't think these tests are useful in anyway; that's an implementation detail for baseUI UIDatePicker